iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/distribution/SourceRewriter.java

```diff
@@ -693,7 +696,7 @@ public List<PlanNode> visitTimeJoin(TimeJoinNode node, DistributionPlanContext c
     return Collections.singletonList(processRawMultiChildNode(node, context, true));
   }
 
-  // Only `visitTimeJoin` and `visitLastQuery` invoke this method
+  // Only `visitFullOuterTimeJoin` and `visitLastQuery` invoke this method
```

Comment:
Maybe `visitLeftOuterTimeJoin` and `visitInnerTimeJoin` can also invoke this method.  I will check this logic later.